cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -271,6 +271,100 @@ __global__ void conditional_join(table_device_view left_table,
   }
 }
 
+template <cudf::size_type block_size, cudf::size_type output_cache_size, bool has_nulls>
+__global__ void conditional_join_semi(
+  table_device_view left_table,
+  table_device_view right_table,
+  join_kind join_type,
+  cudf::size_type* join_output_l,
+  cudf::size_type* current_idx,
+  cudf::ast::detail::expression_device_view device_expression_data,
+  cudf::size_type const max_size)
+{
+  constexpr int num_warps = block_size / detail::warp_size;
```

Comment:
nit: `num_warps` can be computed when launching the kernel

Reply:
So you'll want to just remove the `constexpr` IIUC right? In that case I'll probably want to remove `constexpr` from `conditional_join` as well

Reply:
Actually, I'm happy with the current way to compute #warps. Closing this comment.